src/translate/function.rs

```diff
@@ -73,6 +73,19 @@ pub fn translate_function(sql_function: &SqlFunction) -> Result<Expr> {
         }};
     }
 
+    macro_rules! func_with_two_arg {
+        ($bound: ident, $func: ident) => {{
+            check_len_range(stringify!($func).to_owned(), args.len(), 1, 2)?;
+            let expr = translate_expr(args[0])?;
+            let chars = if args.len() == 1 {
+                None
+            } else {
+                Some(translate_expr(args[1])?)
+            };
+            Ok(Expr::Function(Box::new($bound::$func { expr, chars })))
```

Comment:
It looks a bit ugly, but the best way to pass struct in macro i've found so far.
If there's better way i will fix real quick.

Reply:
Here you can simply try this
```rust
 macro_rules! func_with_two_arg {
        ($func: ident) => {{
            check_len_range(stringify!($func).to_owned(), args.len(), 1, 2)?;
            let expr = translate_expr(args[0])?;
            let chars = if args.len() == 1 {
                None
            } else {
                Some(translate_expr(args[1])?)
            };
            Ok(Expr::Function(Box::new(Function::$func { expr, chars })))
        });
}
```

Reply:
Thank you for simplifying!
All your advices were adjusted and New PR was submitted!